src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exceptions.AuthenticationException;
+import com.dev.cinema.lib.Inject;
+import com.dev.cinema.lib.Service;
+import com.dev.cinema.model.User;
+import com.dev.cinema.service.UserService;
+import com.dev.cinema.util.HashUtil;
+import java.util.Optional;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> user = userService.findByEmail(email);
+        if (user.isPresent() && isPasswordValid(user.get(), password)) {
+            return user.get();
+        }
+        throw new AuthenticationException("Incorrect email or password");
+    }
+
+    @Override
+    public User register(String email, String password) throws AuthenticationException {
+        if (email.isEmpty() || password.isEmpty()) {
+            throw new AuthenticationException("Email or password is empty");
+        }
```

Comment:
let's remove this check, we will validate input later on in controller + you throw AuthenticationException while registration - these are different procedures